Symlink for filenames

aayushKumarJarvis opened the pull request:
This PR is in reference to an issue which we experienced while integrating marathon-logger with **loggly**. Marathon Logger generates a rsyslog configuration file so as to pass logs to `syslog` which can be consumed by loggly. But the problem here was that length of mesos's tasks log file path in a configuration file for `rsyslog` cannot be more than `200` characters 😢 . 

We fixed this issue by creating a symlink for the absolute path for mesos logs for tasks running on a particular machine. This made our life easy here for deploying marathon-logger for all our mesos-marathon based systems. 

cc : @ashwanthkumar

Commit:
Updating tests

Commit:
Fixing indentation errors in tests

aayushKumarJarvis commented:
@brewkode, there were a few fixes in the PR which are now done.

ashwanthkumar commented on glide.lock:
```diff
@@ -34,7 +34,7 @@ imports:
 testImports:
 - name: github.com/davecgh/go-spew
-  version: 5215b55f46b2b919f50a1df0eaa5886afe4e3b3d
+  version: 2df174808ee097f90d259e432cc04442cf60be21
```
@aayushKumarJarvis if this update is not required as part of this change, please revert this.

ashwanthkumar commented on rsyslog-logger.go:
```diff
@@ -21,18 +22,16 @@ const RsyslogTemplate = `
-			Tag="{{ .CleanAppName }}	{{.TaskID}}"
-			statefile="{{ .TaskID }}"
-      Severity="info")
+		File="{{ .WorkDir }}/{{ .FileName }}"
```
@aayushKumarJarvis We always seem to have some space vs tab issue in here. Can you please fix this?

ashwanthkumar commented on rsyslog-logger.go:
@aayushKumarJarvis Can we please extract `fmt.Sprintf("%s/%s", r.WorkDir, taskInfo.TaskID)` into a private function inside `Rsyslog` ? We're using it at another place and it's best if the whole thing is at a single place.

ashwanthkumar commented on task-manager.go:
```diff
@@ -85,42 +86,44 @@ func (t *TaskManager) run() {
-							TaskID:   task.TaskID,
-							CWD:      executor.Directory,
-							FileName: file,
+	//		if task.Hostname == hostname {
```
@aayushKumarJarvis Please revert this section to the one in the master. Check the diff above.

ashwanthkumar committed:
Fixing the intend issue

Also
1. Moving rsyslog tests to it's own file where it truly belongs
2. extracting the symlink dir as a separate method so it's easy to manage

ashwanthkumar committed:
Reverting to use the task-manager's run from master

ashwanthkumar committed:
Reverting the changes on glide.lock since it's not needed
